src/main/java/net/dv8tion/jda/bot/sharding/DefaultShardManagerBuilder.java

```diff
@@ -0,0 +1,573 @@
+     */
+    public DefaultShardManagerBuilder setStatus(final OnlineStatus status)
+    {
+        Checks.check(status != null, "OnlineStatus cannot be null!");
+        Checks.check(status != OnlineStatus.UNKNOWN, "OnlineStatus cannot be unknown!");
+        this.status = status;
+        return this;
+    }
+
+    /**
+     * Sets the token that will be used by the {@link net.dv8tion.jda.bot.sharding.ShardManager ShardManager} instance to log in when
+     * {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#buildAsync() buildAsync()}
+     * or {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder#buildBlocking() buildBlocking()}
+     * is called.
+     *
+     * <p>To get a bot token:
+     * <ol>
+     *     <li>Go to your <a href="https://discordapp.com/developers/applications/me">Discord Applications</a></li>
+     *     <li>Create or select an already existing application</li>
+     *     <li>Verify that it has already been turned into a Bot. If you see the "Create a Bot User" button, click it.</li>
+     *     <li>Click the <i>click to reveal</i> link beside the <b>Token</b> label to show your Bot's {@code token}</li>
+     * </ol>
+     *
+     * @param  token
+     *         The token of the account that you would like to login with.
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the token is either null or empty
+     *
+     * @return The {@link net.dv8tion.jda.bot.sharding.DefaultShardManagerBuilder DefaultShardManagerBuilder} instance. Useful for chaining.
+     */
+    public DefaultShardManagerBuilder setToken(final String token)
+    {
+        Checks.notBlank(token, "token");
+
+        this.token = token;
+        return this;
+    }
+
+    /**
+     * Sets the {@link com.neovisionaries.ws.client.WebSocketFactory WebSocketFactory} that will be used by JDA's websocket client.
+     * This can be used to set things such as connection timeout and proxy.
+     *
+     * @param  factory
+     *         The new {@link com.neovisionaries.ws.client.WebSocketFactory WebSocketFactory} to use.
+     *
+     * @return Returns the {@link net.dv8tion.jda.core.JDABuilder JDABuilder} instance. Useful for chaining.
+     */
+    public DefaultShardManagerBuilder setWebsocketFactory(WebSocketFactory factory)
+    {
+        this.wsFactory = factory;
+        return this;
+    }
+
+    /**
+     * Sets the the backoff in milliseconds that is used as safety between logging into shards.
+     * The final delay will be {@code ratelimit + backoff}, while the ratelimit currently being 5 seconds.
+     * The default backoff is 200 ms. 
+     *
+     * @param  backoff
+     *         The new backoff in ms to use.
+     *
+     * @return Returns the {@link net.dv8tion.jda.core.JDABuilder JDABuilder} instance. Useful for chaining.
+     */
+    public DefaultShardManagerBuilder setLoginBackoff(int backoff)
+    {
+        this.backoff = backoff;
+        return this;
+    }
```

Comment:
This won't be needed if you properly wait until it sends the payload before backing off